transform-data/metricflow
# Update list_tables() in sqlalchemy client to fix Engine.table_names() deprecation warning

**Changes**
This PR updates the list_tables() function in the sqlalchemy client to use `Inspector.get_table_names()` instead of `Engine.table_names()` because it is being deprecated.

This fixed an error when running tests:
```
metricflow/metricflow/sql_clients/sqlalchemy_dialect.py:82: SADeprecationWarning: The Engine.table_names() method is deprecated and will be removed in a future release.  Please refer to Inspector.get_table_names(). (deprecated since: 1.4)
``` 

**Testing:**
1. Run `make test`
2. Observe that the deprecation warning is no longer present.
